### ISSUE TITLE
[HOLD for payment 2023-03-09] [$4000] Opening search using the keyboard shortcut (ctrl+k) while options for adding payment method are visible, options overlap with search.

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#14889]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. navigate to settings ->payments
2. Click on "Add payment method"
3. Press (ctrl+k) on the keyboard

## Expected Result:
options for adding payment method should not appear (Overlap) on search page

## Actual Result:
options for adding payment method appear (Overlap) on search page

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?

- [ ] MacOS / Chrome / Safari

**Version Number:** 1.2.66-0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 

https://user-images.githubusercontent.com/43996225/217110696-562f947d-3279-494e-935d-3bc19be503cd.mp4

https://user-images.githubusercontent.com/43996225/217110709-1b91dd72-e077-4701-9b34-5f537b3813f0.mp4

**Expensify/Expensify Issue URL:**
**Issue reported by:** @Harshdeepjoshi 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1675711618695179

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0106757eb17dcdb76c</li>
        <li>Upwork Job ID: 1622740955154911232</li>
        <li>Last Price Increase: 2023-02-20</li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
Triggered auto assignment to @puneetlath (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~0106757eb17dcdb76c

**Comment by MelvinBot** [comment]
Current assignee @puneetlath is eligible for the External assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @aimane-chnaif (`External`)

**Comment by MelvinBot** [comment]
Current assignee @puneetlath is eligible for the External assigner, not assigning anyone new.

**Comment by allroundexperts** [comment]
## Proposal

### RCA
When a stack navigator route is opened, it does not unmount the previous stack navigator route. Instead, the previous route remains open. In this case, the Popover component of the previous route has a z-Index of `99999` which is higher than the z-index of the new stack navigator screen. Since there is no click involved in opening the search screen and popover menu has a z-index too high, the popover menu keeps on showing up even when the search page is opened.

### Solutions
There can be multiple solutions here depending on the testing effort we want to put into this.

1. We can add another `KeyboardShortcut` listener for `CMD+K` in the `BasePaymentsPage` where when called, it will set the `shouldShowAddPaymentMenu` to `false`. This will cause the popup to close whenever the shortcut is pressed. This is the most failsafe approach as it will not impact any other place.
2. z-Index of the popover can be decreased such that when the new screen opens, the popover menu stays behind it. We'll decrease the z-index for popover only (not the modal). This will require more QA effort to make sure that we did not break anything elsewhere.

**Comment by priyeshshah11** [comment]
## Proposal

### Problem
This issue occurs for all the modals i.e. all modals stay open when any keyboard shortcuts are triggered.

### Solution

We should close all the modals/popovers before opening any new modals or navigating to new routes by calling `Navigation.dismissModal()`. Currently, the places I can think of are before navigating to Search, New Group & Shortcut Modal.

Will post code solution soon.

**Comment by MelvinBot** [comment]
Looks like something related to `react-navigation` may have been mentioned in this issue discussion.

As a reminder, please make sure that all proposals are not workarounds and that any and all attempt to fix the issue holistically have been made before proceeding with a solution. Proposals to change our `DeprecatedCustomActions.js` files should not be accepted.

Feel free to drop a note in [#expensify-open-source](https://expensify.slack.com/archives/C01GTK53T8Q) with any questions.

**Comment by bernhardoj** [comment]
### **Proposal**
Yes, it is indeed happening to all modal, except for FAB. We already have a fix only for FAB and we can use that by moving the part of the logic to the modal component instead. The logic that we need move to the modal component is when the screen focus is changed, we then close the modal. We can't move the drawer state logic from FAB because not all modal is inside a drawer navigator, so that logic is still there on the FAB component.

https://github.com/Expensify/App/blob/26fdacee7e13cb4c9f1d563ac224658c3d65b25b/src/pages/home/sidebar/SidebarScreen/FloatingActionButtonAndPopover.js#L80-L108

```diff
diff --git a/src/components/Modal/BaseModal.js b/src/components/Modal/BaseModal.js
index 1b58579f93..c3bf01e25e 100644
--- a/src/components/Modal/BaseModal.js
+++ b/src/components/Modal/BaseModal.js
@@ -10,9 +10,13 @@ import {propTypes as modalPropTypes, defaultProps as modalDefaultProps} from './
 import * as Modal from '../../libs/actions/Modal';
 import getModalStyles from '../../styles/getModalStyles';
 import variables from '../../styles/variables';
+import compose from '../../libs/compose';
+import withNavigationFocus, {withNavigationFocusPropTypes} from '../withNavigationFocus';
+import withNavigationFallback from '../withNavigationFallback';
 
 const propTypes = {
     ...modalPropTypes,
+    ...withNavigationFocusPropTypes,
 
     /** The ref to the modal container */
     forwardedRef: PropTypes.func,
@@ -31,6 +35,13 @@ class BaseModal extends PureComponent {
     }
 
     componentDidUpdate(prevProps) {
+        if (this.didScreenBecomeInactive(prevProps)) {
+            // Hide popover manually when other pages are opened using shortcut key
+            if (this.props.isVisible) {
+                this.props.onClose();
+            }
+        }
+
         if (prevProps.isVisible === this.props.isVisible) {
             return;
         }
@@ -43,6 +54,22 @@ class BaseModal extends PureComponent {
         this.hideModal(this.props.isVisible);
     }
 
+    /**
+     * Check if page status changed from active to inactive.
+     * Used to close already opened popover when open any other pages (i.e. Press Command + K on web).
+     *
+     * @param {Object} prevProps
+     * @return {Boolean}
+     */
+    didScreenBecomeInactive(prevProps) {
+        // When any other page is opened over the page
+        if (!this.props.isFocused && prevProps.isFocused) {
+            return true;
+        }
+
+        return false;
+    }
+
     /**
      * Hides modal
      * @param {Boolean} [callHideCallback=true] Should we call the onModalHide callback
@@ -166,7 +193,10 @@ class BaseModal extends PureComponent {
 BaseModal.propTypes = propTypes;
 BaseModal.defaultProps = defaultProps;
 
-export default React.forwardRef((props, ref) => (
+export default compose(
+    withNavigationFallback,
+    withNavigationFocus,
+)(React.forwardRef((props, ref) => (
     // eslint-disable-next-line react/jsx-props-no-spreading
     <BaseModal {...props} forwardedRef={ref} />
-));
+)));
diff --git a/src/pages/home/sidebar/SidebarScreen/FloatingActionButtonAndPopover.js b/src/pages/home/sidebar/SidebarScreen/FloatingActionButtonAndPopover.js
index 6fe0447ef4..c999e68275 100644
--- a/src/pages/home/sidebar/SidebarScreen/FloatingActionButtonAndPopover.js
+++ b/src/pages/home/sidebar/SidebarScreen/FloatingActionButtonAndPopover.js
@@ -99,11 +99,6 @@ class FloatingActionButtonAndPopover extends React.Component {
             return true;
         }
 
-        // When any other page is opened over LHN
-        if (!this.props.isFocused && prevProps.isFocused) {
-            return true;
-        }
-
         return false;
     }
 
```

However, it does not cover the case for keyboard shortcut modal. When open the keyboard shortcut from Settings > About and we press ctrl+k for example, the keyboard shortcut modal will not be closed because the screen focus where the keyboard shortcut modal resides does not change. That is because `KeyboardShortcutModal` is placed inside `SidebarScreen` component.

When we open Settings > About, the sidebar screen `isFocused` will be changed to false already and when we open another page from the shortcut, the focus state will not be changed. So, we need to manually hide the keyboard shortcut modal when the new group/chat shortcut is executed.

https://github.com/Expensify/App/blob/26fdacee7e13cb4c9f1d563ac224658c3d65b25b/src/pages/home/sidebar/SidebarScreen/BaseSidebarScreen.js#L50-L69

```diff
diff --git a/src/libs/Navigation/AppNavigator/AuthScreens.js b/src/libs/Navigation/AppNavigator/AuthScreens.js
index 58fae5f8b4..bf44befe49 100644
--- a/src/libs/Navigation/AppNavigator/AuthScreens.js
+++ b/src/libs/Navigation/AppNavigator/AuthScreens.js
@@ -30,6 +30,7 @@ import defaultScreenOptions from './defaultScreenOptions';
 import * as App from '../../actions/App';
 import * as Download from '../../actions/Download';
 import * as Session from '../../actions/Session';
+import * as KeyboardShortcutsActions from '../../actions/KeyboardShortcuts';
 
 let currentUserEmail;
 Onyx.connect({
@@ -114,9 +115,11 @@ class AuthScreens extends React.Component {
         // based on the key modifiers pressed and the operating system
         this.unsubscribeSearchShortcut = KeyboardShortcut.subscribe(searchShortcutConfig.shortcutKey, () => {
             Navigation.navigate(ROUTES.SEARCH);
+            KeyboardShortcutsActions.hideKeyboardShortcutModal();
         }, searchShortcutConfig.descriptionKey, searchShortcutConfig.modifiers, true);
         this.unsubscribeGroupShortcut = KeyboardShortcut.subscribe(groupShortcutConfig.shortcutKey, () => {
             Navigation.navigate(ROUTES.NEW_GROUP);
+            KeyboardShortcutsActions.hideKeyboardShortcutModal();
         }, groupShortcutConfig.descriptionKey, groupShortcutConfig.modifiers, true);
     }
 
```

**Result**

https://user-images.githubusercontent.com/50919443/217141687-9473ba52-5227-4767-a463-b5130aa88582.mov

**Comment by onceLearner** [comment]
# Proposal
## Problem 
When a popover menus are opened,  triggering the search with keyboard shortcuts keep them opened.
## Solution
Attach Keyboard listeners to close the popover menu once the keyboard is triggered.
Changes will be made to `src/components/PopoverMenu/index.js`.
So joining @allroundexperts   proposal while adding more customization.

<img width="981" alt="image" src="https://user-images.githubusercontent.com/41970812/217147487-e20107c6-176e-4022-8793-c8ba66349e20.png">

Then a method to remove the listeners:
<img width="815" alt="image" src="https://user-images.githubusercontent.com/41970812/217148568-d34961fa-483e-4af6-a689-d88aa0d19cb8.png">


## Screen

https://user-images.githubusercontent.com/41970812/217148373-62656def-b5be-42b5-9abe-196ee9d42c92.mov

**Comment by Snehal-Techforce** [comment]
### **Proposal**

**Problem:**
Modals stay open when any keyboard shortcuts are triggered.

**Root Cause:** In the _AuthScreens.js_ file, the code is missing to dismiss the previously opened modal in the keyboard shortcut listeners.

![image](https://user-images.githubusercontent.com/123076498/217787979-d8441259-0900-4fe5-9c98-fba5aafc1ed9.png)

**Solution:** _**Navigation.dismissModal()**_ needs to add on below place in _AuthScreens.js_ file, in order to close old modal when new modal arise.

![image](https://user-images.githubusercontent.com/123076498/217788705-be017671-195a-4c27-8205-5838dd0db8fa.png)

**Comment by situchan** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Press Ctrl+K on keyboard when any popover modal shows. Search page opens but it's behind popover and not active until popover is closed manually.

### What is the root cause of that problem?
There are 3 keyboard shortcuts to open new page: Search, New group, Shortcut modal.
Since keyboard shortcut works everywhere, this also works when any modal is open.
Modal is always top priority in view hierarchy so new page which is opened later is still behind modal.
We didn't handle this case from the beginning when keyboard shortcut is implemented.

### What changes do you think we should make in order to solve the problem?
The solution is to close any popover open which prevents new page being being active.

**1. Close any popover when Search page (Ctrl+K) or New group page (Ctrl+Shift+K) is opening**
The ideal component to handle this case is `PopoverMenu`.
There's already existing code of handling `ENTER` shortcut key in `attachKeyboardListener` when visible, `removeKeyboardListener` when invisible.
https://github.com/Expensify/App/blob/39d9b660e387b28688a221b976f0fbf52873cc67/src/components/PopoverMenu/index.js#L59-L75
We can use similar approach for `NEW_GROUP`, `SEARCH` shortcuts. So in subscribe callback, just call `this.props.onClose()` to close modal.

**2. Close any popver when Shortcut modal is opening**
We can use same approach as 1. but there's already existing flag saved in Onyx so we can use this without subscribing another listener: `isShortcutsModalOpen: {key: ONYXKEYS.IS_SHORTCUTS_MODAL_OPEN},`
https://github.com/Expensify/App/blob/39d9b660e387b28688a221b976f0fbf52873cc67/src/components/PopoverMenu/index.js#L34-L44
We can catch shortcut modal opened state in `componentDidUpdate` of `PopoverMenu` and call `this.props.onClose()` to close popover.

**3. Close shortcut modal when Search page or New group page is opening**
SEARCH, NEW_GROUP shortcut key event listeners are added in `AuthScreens` so we can use this to hide keyboard shortcut modal.
https://github.com/Expensify/App/blob/39d9b660e387b28688a221b976f0fbf52873cc67/src/libs/Navigation/AppNavigator/AuthScreens.js#L116-L121
Just call `KeyboardShortcutsActions.hideKeyboarShortcutModal()` in subscribe callbacks.

**Demo:**

https://user-images.githubusercontent.com/108292595/218339553-e1c07362-40ee-43c8-b01f-055f0ed74962.mov

**Comment by priyeshshah11** [comment]
@aimane-chnaif any updates/feedback on the proposals above?

**Comment by allroundexperts** [comment]
> @allroundexperts's [proposal](https://github.com/Expensify/App/issues/14889#issuecomment-1419992776) is limited to Payments page. It should be general solution. i.e. when click `+` on composer

@aimane-chnaif Can you please list down the case where this fails in particular? Pressing a `+` on composer and then opening side menu doesn't seem to cause any problem. Can you please elaborate?

**Comment by allroundexperts** [comment]
@aimane-chnaif is this a real world scenario? I mean, users on mobile won't be able to use the shortcuts anyway.

> > @aimane-chnaif Can you please list down the case where this fails in particular? Pressing a `+` on composer and then opening side menu doesn't seem to cause any problem. Can you please elaborate?
> 
> 
> 
> @allroundexperts 👇 
> 
> 
> 
> https://user-images.githubusercontent.com/96077027/218563477-7db0e9c5-1a09-4f36-9cfe-2f2b1c615448.mov
> 
>

**Comment by allroundexperts** [comment]
> > @aimane-chnaif is this a real world scenario? I mean, users on mobile won't be able to use the shortcuts anyway.
> 
> 
> 
> @allroundexperts This is not mobile but web with small width

Exactly. Even on mweb these shortcuts aren't available. Unless someone connects a keyboard with mobile?

**Comment by allroundexperts** [comment]
> > > @aimane-chnaif is this a real world scenario? I mean, users on mobile won't be able to use the shortcuts anyway.
> > 
> > 
> > 
> > @allroundexperts This is not mobile but web with small width
> 
> Exactly. Even on mweb these shortcuts aren't available. Unless someone connects a keyboard with mobile?

You're hitting mobile screen breakpoint and trying to use keyboard there.

**Comment by allroundexperts** [comment]
> @aimane-chnaif is this a real world scenario? I mean, users on mobile won't be able to use the shortcuts anyway.
> 
> > > @aimane-chnaif Can you please list down the case where this fails in particular? Pressing a `+` on composer and then opening side menu doesn't seem to cause any problem. Can you please elaborate?
> > 
> > 
> > 
> > @allroundexperts 👇 
> > 
> > 
> > 
> > https://user-images.githubusercontent.com/96077027/218563477-7db0e9c5-1a09-4f36-9cfe-2f2b1c615448.mov
> > 
> > 
> 
> 

If this is really a bummer, we can add this shortcut listener to the + popover as well. That will fix it for your mentioned case. We can also move the listener inside the popover component as well to make it more generic. Let me know if this works for you.

**Comment by allroundexperts** [comment]
> Another example:
> 
> 
> 
> https://user-images.githubusercontent.com/96077027/218567432-f7bb4278-2304-44ec-a527-a511fe64898d.mov
> 
> 
> 
> 

@aimane-chnaif have you applied my changes correctly? This works with my changes.

**Comment by allroundexperts** [comment]
> ```diff
>  this.unsubscribeSearchShortcut = KeyboardShortcut.subscribe(searchShortcutConfig.shortcutKey, () => {
> +            this.hideAddPaymentMenu();
> +        }, searchShortcutConfig.descriptionKey, searchShortcutConfig.modifiers, true, true, undefined, false);
> +        this.unsubscribeGroupShortcut = KeyboardShortcut.subscribe(groupShortcutConfig.shortcutKey, () => {
> +            this.hideAddPaymentMenu();
> +        }, groupShortcutConfig.descriptionKey, groupShortcutConfig.modifiers, true, true, undefined, false);
>      }
> ```

@aimane-chnaif Yea, you're right about this. How about moving this logic inside the popover menu component itself? That will solve it everywhere.

**Comment by allroundexperts** [comment]
You've actually done something similar here as well:
https://github.com/Expensify/App/pull/12031

**Comment by bernhardoj** [comment]
@aimane-chnaif Can you make sure `withNavigationFallback` HOC is being used on `BaseModal` and put before `withNavigationFocus`? I think the order also matters.

**Comment by s77rt** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Opening other pages overlaps with opened modal(s).

### What is the root cause of that problem?
We don't hide open modals navigating to other pages.

### What changes do you think we should make in order to solve the problem?
Hide open modals on navigation change. The `react-navigation` listener is not available for all modals so using `withNavigation` HOC won't work. Instead we will have to manage those events manually by dispatching them [here](https://github.com/Expensify/App/blob/d42e443d35762f15d1ebd3181bbc17636dff3d98/src/libs/Navigation/Navigation.js#L160) and consuming them [here](https://github.com/Expensify/App/blob/e31cb5349e802a0c7386e6d171834344f63322a9/src/components/Modal/BaseModal.js#L26).

To dispatch events we will make use of [dispatchEvent](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/dispatchEvent) (Web Only).

### What alternative solutions did you explore? (Optional)
- Do nothing. I don't think we should fix this especially since it's related to keyboard navigation (which we are currently holding)

**Comment by puneetlath** [comment]
> I don't think we should fix this especially since it's related to keyboard navigation

Did you mean it's related to app navigation? If so, I could see that argument. What do you think @aimane-chnaif?

**Comment by situchan** [comment]
I have an updated proposal which applies to all modals, not only popovers.
Closing modal on navigation state change is not a reliable solution. And apart from shortcut keys, there's no way of changing navigation state while showing modal, and thus, no need to subscribe navigation changes in every modal (for performance and optimization).
Finally I think the solution below is simple and reasonable:

**1. Close any popover when Search page (Ctrl+K) or New group page (Ctrl+Shift+K) is opening**
- define `closeModal` function as a global variable (/libs/actions/Modal.js is the best place I think)
- set `onClose` value of Modal prop to this variable only when modal is open. Set to null when closed
- call `closeModal` function upon keyboard shortcuts of search, new group (event listeners are defined in AuthScreens.js) -> this calls `onClose` of Modal -> modal closes

[Demo](https://github.com/situchan/Expensify/commit/12b46d85052038aa055d89da42378e49fad930b6) (I understand code diff is not recommended but just for testing purpose)

**2. Close any popover when Shortcut modal is opening**
This is already done in https://github.com/Expensify/App/pull/13956 but has issue I reported [here](https://github.com/Expensify/App/issues/13871#issuecomment-1436115158) with the solution.
Alternatively, we can use same approach as in **1.** (in shortcut subscribe event listener defined in KeyboardShortcutsModal.js) -> I think this is simpler solution and recommended. cc: @pecanoro 

**3. Close shortcut modal when Search page or New group page is opening**
Shortcut modal is also Modal so this is already done in **1.**

**Comment by s77rt** [comment]
@aimane-chnaif Yes, try with the report actions context menu.
1. Hover over any message
2. Right click to open the context menu
3. Press Ctrl+K

I think none of the above solutions handles this case (excluding the overkill solution of dispatching events)

**Comment by priyeshshah11** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Avoid the popovers from being displayed over the most recently opened modal.

### What is the root cause of that problem?
The root cause of the problem is that we don't dismiss those popovers before opening the new modal thus it appears over it.

### What changes do you think we should make in order to solve the problem?
We should rename & use the `isShortcutsModalOpen` key to keep track of the state of all such modals where we want other popovers to be dismissed before opening these modals.

**Comment by MelvinBot** [comment]
@puneetlath @aimane-chnaif this issue was created 2 weeks ago. Are we close to approving a proposal? If not, what's blocking us from getting this issue assigned? Don't hesitate to create a thread in #expensify-open-source to align faster in real time. Thanks!

**Comment by priyeshshah11** [comment]
@aimane-chnaif why do you think adding a new ONYK key is not a good/better solution when we already use ONYX keys for something as simple as displaying a particular modal e.g. Keyboard shortcuts modal

**Comment by situchan** [comment]
Using onyx value to get modal open/closed status is not recommended at all.
We don't guarantee that this weird issue won't happen on all modals.

https://user-images.githubusercontent.com/108292595/220308663-76cfc37b-5491-4b65-b3eb-5b8c61048c9c.mov


2 issues in this video:
- Currently open modal is still open after refreshing page or visit same url on another tab on same browser (Is this a bug?)
- Some contents of keyboard shortcut modal are hidden until refresh

**Comment by priyeshshah11** [comment]
@situchan yes that looks like a weird bug, but I still don't understand what does that have to do with using onyx or not? Are you saying that using onyx is the cause of this weird issue?

**Comment by priyeshshah11** [comment]
@aimane-chnaif I got your point about the shortcuts modal open key, thanks for explaining.

> @priyeshshah11 If we introduce new onyx key for every modal, I am curious what will be key and what will be value?

The new key can be called isClosableModalOrPopoverOpen or some better word than close, the value should simply be a boolean. And yes, you're correct it should be applied to both BaseModal & Popover as per my latest proposal.

> What do you mean? Can you share video? Also, don't post proposal similar to previous ones.

I am talking about this

<img width="530" alt="Screen Shot 2023-02-22 at 1 26 16 AM" src="https://user-images.githubusercontent.com/38547776/220371851-e5c13b91-9e4b-4a28-bb6a-e6c935fa9ed1.png">

With the new approach a lot of proposals are going to be similar with some differences, so what do you want me to do when I see issues/improvements with current proposals? we're definitely not supposed to edit proposals once posted.

**Comment by puneetlath** [comment]
I think @situchan's proposal sounds good too. Let's do it.

**Comment by MelvinBot** [comment]
📣 @situchan You have been assigned to this job by @puneetlath!
Please apply to this job in Upwork and leave a comment on the Github issue letting us know when we can expect a PR to be ready for review 🧑‍💻
Keep in mind: [Code of Conduct](https://github.com/Expensify/App/blob/main/CODE_OF_CONDUCT.md) | [Contributing](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md) 📖

**Comment by MelvinBot** [comment]
@puneetlath, @aimane-chnaif, @situchan Whoops! This issue is 2 days overdue. Let's get this updated quick!

**Comment by MelvinBot** [comment]
`Reviewing` label has been removed, please complete the "BugZero Checklist".

**Comment by MelvinBot** [comment]
The solution for this issue has been :rocket: deployed to production :rocket: in version 1.2.77-4 and is now subject to a 7-day regression period :calendar:. Here is the list of pull requests that resolve this issue:

- https://github.com/Expensify/App/pull/15413


If no regressions arise, payment will be issued on 2023-03-09. :confetti_ball:

After the hold period is over and BZ checklist items are completed, please complete any of the applicable payments for this issue, and check them off once done.
- [ ] External issue reporter
- [ ] Contributor that fixed the issue
- [ ] Contributor+ that helped on the issue and/or PR

As a reminder, here are the bonuses/penalties that should be applied for any External issue:
- Merged PR within 3 business days of assignment - 50% **bonus**
- Merged PR more than 9 business days after assignment - 50% **penalty**

**Comment by MelvinBot** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@aimane-chnaif / @puneetlath] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@aimane-chnaif / @puneetlath] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@aimane-chnaif / @puneetlath] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@situchan] Propose [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
- [ ] [@puneetlath] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by puneetlath** [comment]
@situchan @Harshdeepjoshi can you please apply to the Upwork job? https://www.upwork.com/jobs/~0106757eb17dcdb76c

@aimane-chnaif can you please help with the checklist? 

Thanks!

**Comment by puneetlath** [comment]
@Harshdeepjoshi @situchan sent you hiring offers! 

@aimane-chnaif paid!

**Comment by puneetlath** [comment]
All paid. Thanks everyone!
